linux-kernel config: disable DEBUG_INFO_REDUCED

martinetd opened the pull request:
###### Description of changes

Linux's aarch64 defconfig has been updated in 5.13 to enable "reduced" debug infos (upstream commit [ed938a4bfc58](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=ed938a4bfc58cd86eddcb4b1c9d582a435ffcf7b) ("arm64: defconfig: Use DEBUG_INFO_REDUCED"), but that commits locks DEBUG_INFO_BTF as noticed in #175467

This disables it back which should fix bpftrace usage of BTF not working on newer kernels.


Note I don't have the resources to rebuild, so didn't check. I'll try to trigger a nixosTests.bpf run on ofborg with this to validate.
 
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @Mic92

martinetd commented:
@ofborg build nixosTests.bpf

Mic92 commented on pkgs/os-specific/linux/kernel/common-config.nix:
```diff
@@ -41,6 +41,9 @@ let
       DEBUG_INFO_DWARF_TOOLCHAIN_DEFAULT = whenAtLeast "5.18" yes;
+      # Reduced debug info conflict with BTF and have been enabled in
+      # aarch64 defconfig since 5.13
+      DEBUG_INFO_REDUCED        = no;
```
For linux 4.19, I get: `error: unused option: DEBUG_INFO_REDUCED` can you add a lower bound to avoid breaking older kernels?

martinetd replied:
Hmm, I didn't because the option appeared in v2.6.36 (! https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=d6f4ceb796ebf1a8c8a9ad4a8ea0d181aaec7de6 )

Since I'm not sure why it's not used for you and we only care about 5.13+ kernels I've just arbitrarily made that the limit, but it's a bit weird. Also, why didn't the config system complain about DEBUG_INFO_BTF not being asked when it was masked by DEBUG_INFO_REDUCED?.. I don't quite get this configuration system, sorry.

martinetd commented:
@ofborg build nixosTests.bpf

(triggering a new build for principle but it worked the first time)

martinetd committed:
linux-kernel config: disable DEBUG_INFO_REDUCED

Linux's aarch64 defconfig has been updated in 5.13 to enable "reduced"
debug infos (upstream commit ed938a4bfc58 ("arm64: defconfig: Use
DEBUG_INFO_REDUCED"), but that commits locks DEBUG_INFO_BTF as noticed
in #175467

This disables it back which should fix bpftrace usage of BTF not working
on newer kernels.

martinetd commented:
ping -- what should we do about it?
(I just rebased to fix the merge conflict)
